### ISSUE TITLE
Horizontal and Vertical Scrolling is not working.

### ISSUE THREAD

**Issue by chnouman** [pchmn/MaterialChipsInput#18]
I'm trying to show this with property max rows: but it is not providing me a scrolling.Please help me

**Comment by MoeHamoud** [comment]
wrap it in a scrollview?

**Comment by pchmn** [comment]
I don't understand well what's not working.

When you use `maxRows` like this, it is not working ?
```
<com.pchmn.materialchips.ChipsInput
        android:id="@+id/chips_input"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        app:hint="Enter a name"
        app:maxRows="3" />
```

Because if you look at the [sample app](https://github.com/pchmn/MaterialChipsInput/blob/master/sample/src/main/res/layout/activity_contact_list.xml), the `maxRows` attribute is successfully used.

**Comment by chnouman** [comment]
sorry actually i was in scrollview in scrollview maxRows property was not working.

**Comment by chnouman** [comment]
yes @MoeHamoud  it was in scrollview.

**Comment by chnouman** [comment]
@MoeHamoud  @pchmn  still back button is not working when i try to remove last chip from chips.

**Comment by pchmn** [comment]
The bug occurred just by using the `ChipsInput` view in a `ScrollView` in your xml layout ?

Can you provide the code that is responsible of the bug ? To reproduce it.

**Comment by chnouman** [comment]
yes it only occurs when we are in Scrollview i changed my layout now its ok. but back button on key board don't remove last chip please help me.

**Comment by pchmn** [comment]
I will test it and try to reproduce the bug. 
I will return to you when I found a fix

**Comment by chnouman** [comment]
ok

**Comment by chnouman** [comment]
Hy can you please tell me how to add chip automatically or programmatcaly without selecting from drop down list.

**Comment by MoeHamoud** [comment]
@chnouman I'm not sure you even need this library for such functionality. 

I've done this before and if what you want is what I think it is then this is how I would go about it.

- Create a layout file (call it, `textview_chip.xml`, for example) with a `FrameLayout` as the root element and a `TextView` inside it
- Style the `TextView` or even the `FrameLayout` so it looks like a chip. I'm not going to go into detail about this but there are resources out there to help you by using an **xml drawable** and setting it as the background of the `TextView` or `FrameLayout`
- You could then create a customView group (call it, `CustomChipView` for example) that extends `FrameLayout` and inflate the `textview_chip.xml`
- From there, you can write public functions inside the `CustomChipView`, for instance, `setChipText()`, `setChipAvatar()`, etc. Depending on what data you want to present to the user
- You could then instantiate your `CustomChipView` and use, for example, `linearLayout.addView(CustomChipView)` to add it to your layout programmatically. You could then set an `onClick listener` on the `CustomChipView` which would remove it from the `linearLayout`.

This is one way that I've "faked" chips.

You can also use the same approach inside a `recyclerView`. You wouldn't need to create a customView group to inflate the `textview_chip.xml` layout but you would just inflate it inside the recyclerView's adapter.

Just another option for you, maybe someone else can help you in terms of using this library to achieve the functionality you're after.

**Comment by chnouman** [comment]
hy sorry i was unable to clearly explain my problem actually the problem is in my scenario the user go to contacts activity and mark some contacts and say ok and come back to parent activity now on parent activity  i want to put all  those selected contacts in  chips.

**Comment by MoeHamoud** [comment]
Well you need to store the selected contacts in a list and then pass it back to the parent activity. From there, loop through your list of contacts and add them as chips to your view.

See: https://stackoverflow.com/questions/11340776/passing-a-list-from-one-activity-to-another

**Comment by chnouman** [comment]
@MoeHamoud let me know how to add chips to my view thank you.

**Comment by MoeHamoud** [comment]
@chnouman its in the docs

```
chipsInput.addChip(ChipInterface chip);
// or
chipsInput.addChip(Object id, Drawable icon, String label, String info);
// or
chipsInput.addChip(Drawable icon, String label, String info);
// or
chipsInput.addChip(Object id, Uri iconUri, String label, String info);
// or
chipsInput.addChip(Uri iconUri, String label, String info);
// or
chipsInput.addChip(String label, String info);
```

**Comment by chnouman** [comment]
Thank you dear

**Comment by dkgupta2501** [comment]
inside scrollview, i am using chipInput ,its working perfectly but when i scroll the layout from from top, its not scrolling, what should i do for that.

**Comment by dkgupta2501** [comment]
can anyone fixed my problem, it will highly appreciated.

**Comment by jahanvi-droid** [comment]
ChipInput inside scrollview/nested scrollview doesn't working. If anyone has solution. Please share it.

**Comment by sajadsoft1** [comment]
ChipInput not working in scroll view , chips adapter not show

**Comment by nicolasrodf** [comment]
You can clear focus from chip_input listening scroll status of scrollView:

scrollView.getViewTreeObserver().addOnScrollChangedListener(new OnScrollChangedListener() {
@override
public void onScrollChanged() {
chipsInput.clearFocus();
}
});

This clear focus and you can scrolling
